### PR TITLE
[TD-6140][New Feature] cast functions in SQL

### PR DESCRIPTION
**Pull request by dapan1121**

### CODE REVIEW

File: src/client/src/tscGlobalmerge.c [hunk 1]
```diff
@@ -605,8 +605,10 @@ static void doMergeResultImpl(SMultiwayMergeInfo* pInfo, SQLFunctionCtx *pCtx, i
     if (functionId < 0) {
       SUdfInfo* pUdfInfo = taosArrayGet(pInfo->udfInfo, -1 * functionId - 1);
       doInvokeUdf(pUdfInfo, &pCtx[j], 0, TSDB_UDF_FUNC_MERGE);
-    } else {
+    } else if (!TSDB_FUNC_IS_SCALAR(functionId)){
```
Comment (xiao-yu-wang) [inline src/client/src/tscGlobalmerge.c]: The modification here is basically consistent with the original logic. Proposed change to:
`else {
  assert(!TSDB_FUNC_IS_SCALAR(functionId))
  aAggs[functionId].mergeFunc(&pCtx[j]);
}`

File: src/client/src/tscGlobalmerge.c [hunk 2]
```diff
@@ -621,8 +623,10 @@ static void doFinalizeResultImpl(SMultiwayMergeInfo* pInfo, SQLFunctionCtx *pCtx
     if (functionId < 0) {
       SUdfInfo* pUdfInfo = taosArrayGet(pInfo->udfInfo, -1 * functionId - 1);
       doInvokeUdf(pUdfInfo, &pCtx[j], 0, TSDB_UDF_FUNC_FINALIZE);
-    } else {
+    } else if (!TSDB_FUNC_IS_SCALAR(functionId)){
```
Comment (xiao-yu-wang) [inline src/client/src/tscGlobalmerge.c]: ditto

File: src/client/src/tscGlobalmerge.c [hunk 3]
```diff
@@ -661,8 +665,11 @@ static void doExecuteFinalMerge(SOperatorInfo* pOperator, int32_t numOfExpr, SSD
           if (pCtx[j].functionId < 0) {
             continue;
           }
-
-          aAggs[pCtx[j].functionId].init(&pCtx[j], pCtx[j].resultInfo);
+          if (TSDB_FUNC_IS_SCALAR(pCtx[j].functionId)) {
```
Comment (xiao-yu-wang) [inline src/client/src/tscGlobalmerge.c]: ditto

File: src/client/src/tscGlobalmerge.c [hunk 4]
```diff
@@ -902,8 +912,11 @@ SSDataBlock* doGlobalAggregate(void* param, bool* newgroup) {
             clearOutputBuf(&pAggInfo->binfo, &pAggInfo->bufCapacity);
             continue;
           }
-
-          aAggs[pCtx->functionId].init(pCtx, pCtx->resultInfo);
+          if (!TSDB_FUNC_IS_SCALAR(pCtx->functionId)) {
```
Comment (xiao-yu-wang) [inline src/client/src/tscGlobalmerge.c]: ditto

File: src/common/inc/texpr.h [hunk 5]
```diff
@@ -74,8 +125,19 @@ typedef struct tExprNode {
...
+
     tVariant           *pVal;
+
+    struct {
+      int16_t functionId;
+      int32_t numChildren;
+      struct tExprNode **pChildren;
+    } _func;
+
+    TAOS_FIELD        *pType;
```
Comment (xiao-yu-wang) [inline src/common/inc/texpr.h:137]: TAOS_FIELD is defined in taos.h of the client API. Not recommended in internal implementations.

File: src/client/src/tscGlobalmerge.c (hunk 1 above)
Comment (shenglian-zhou) [reply src/client/src/tscGlobalmerge.c]: yes. it will be modified

File: src/client/src/tscGlobalmerge.c (hunk 2 above)
Comment (shenglian-zhou) [reply src/client/src/tscGlobalmerge.c]: yes it will be modified

File: src/client/src/tscGlobalmerge.c (hunk 3 above)
Comment (shenglian-zhou) [reply src/client/src/tscGlobalmerge.c]: yes. it will be modified

File: src/client/src/tscGlobalmerge.c (hunk 4 above)
Comment (shenglian-zhou) [reply src/client/src/tscGlobalmerge.c]: yes. it will be modified

**Pull request merged**